### PR TITLE
no-commit-to-branch: Add 'main' to branches blocked by default

### PR DESCRIPTION
**Pull request by ndevenish**
It's becoming more common to have the primary git branch be called `main` instead of `master` now. With GitHub now making it the [default](https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/) and making it [easier](https://github.com/github/renaming) to transition across, we've started noticing it more in other repositories, and started transferring our repositories over to it.

However, a stage that we missed in several cases was updating the pre-commit hook for `no-commit-to-branch` to block the new name, as we had relied on the default behaviour before.

This patch updates the default behaviour so both `master` and `main` are blocked by default.

~Along with this, I found that all of the tests for this hook failed by default on my machine - as part of moving my muscle memory I had set git's global `init.defaultBranch=main`. This only seems to exist from 2.28, while pre-commit lists 2.24 as a requirement for tests - but git seems to ignore (or set but not use) unrecognised configurations, so this should be safe for all targets.~

~_Side note: I couldn't find any guidelines for what formatting you want in this repo, and flake8 complained about the now-longer git call in the test configuration, so I wrote it in black-style._~

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -6,5 +6,12 @@
...
     git_dir = tmpdir.join('gits')
-    cmd_output('git', 'init', '--', str(git_dir))
+    cmd_output(
+        'git',
+        '-c',
+        'init.defaultBranch=master',
+        'init',
+        '--',
+        str(git_dir),
+    )
```
Comment (asottile) [inline tests/conftest.py:16]: why is this changed, this has nothing to do with your patch

Comment (ndevenish) [reply tests/conftest.py:16]: From the PR:
> Along with this, I found that all of the tests for this hook failed by default on my machine - as part of moving my muscle memory I had set git's global init.defaultBranch=main. This only seems to exist from 2.28, while pre-commit lists 2.24 as a requirement for tests - but git seems to ignore (or set but not use) unrecognised configurations, so this should be safe for all targets.

This should have explicitly added the words “I have changed the test to force the configuration to `master` so that it is no longer implicitly assumed”. It seems reasonable to permit the tests to be run in an environment which is likely to exist on the machines it affects, but I’m happy for this not to be the case and separate/remove.

Comment (asottile) [reply tests/conftest.py:16]: could you make a separate pr, it's unrelated to this change

Comment (asottile) [reply tests/conftest.py:16]: or point out which test you're referring to?  it might be better to just change that test

Comment (ndevenish) [reply tests/conftest.py:16]: > all of the tests for this hook

Comment (ndevenish) [reply tests/conftest.py:16]: Although I'm happy to just pull this out

**Comment by ndevenish** [comment]
I've removed the change to the test configuration that caused all of the `no-commit-to-branch` tests to fail on my machine.

**Review by asottile — APPROVED** [review]
if you want to make another PR which makes these tests work independent of the default configured branch that'd be cool
